src/Serilog.Sinks.PeriodicBatching/Sinks/PeriodicBatching/PeriodicBatchingSink.cs

```diff
@@ -52,27 +52,16 @@ public abstract class PeriodicBatchingSink : ILogEventSink, IDisposable
         /// </summary>
         /// <param name="batchSizeLimit">The maximum number of events to include in a single batch.</param>
         /// <param name="period">The time to wait between checking for event batches.</param>
-        protected PeriodicBatchingSink(int batchSizeLimit, TimeSpan period)
+        /// <param name="queueLimit">Maximum number of events in the queue.</param>
+        protected PeriodicBatchingSink(int batchSizeLimit, TimeSpan period, int? queueLimit = null)
```

Comment:
The parameter type change will be binary-breaking for inheritors.

Reply:
How about the new update, where the original constructors are maintained but a new one added. 

This has the added benefit that a call to the original constructor specifying the queueLimit now doesn't create a queue with no limit, then immediately replace it with one that has a limit.